### PR TITLE
Refactor RemoteAPI users to share more code [AP-2069]

### PR DESCRIPTION
**Pull request by chouquette**
### What does this PR do?

This PR refactors the usage of RemoteAPI to use a single common implementation of `make_request` instead of having a duplicated version with minor differences in all subclasses.

### Motivation

AP-2069

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by chouetz — APPROVED** [review]

File: tasks/libs/common/github_api.py [hunk 1]
```diff
@@ -76,27 +76,11 @@ def get_milestone_by_name(self, milestone_name):
...
-        return self.request(
-            path=path,
-            headers=headers,
-            data=data,
-            json_input=False,
-            json_output=json_output,
-            stream_output=False,
-            method=method,
-        )
+    def auth_headers(self):
```
[the remark this answers is not in the document]
Comment (chouquette) [reply tasks/libs/common/github_api.py:79]: I dislike the idea of using API specific headers as the default, but we could reach a similar result by having GithubWorkflow inheriting from GithubAPI (I'm not sure about the GithubApp class yet)

[the remark this answers is not in the document]
Comment (chouetz) [reply tasks/libs/common/github_api.py:79]: ok then no need to do more, on my side I'm not fan of spending time on our own implementation of such api

[the remark this answers is not in the document]
Comment (chouquette) [reply tasks/libs/common/github_api.py:79]: Agreed, I just saw your ticket to switch to an official implementation, I might take a look at it :)

**Comment by chouquette** [comment]
Closing in favor of https://github.com/DataDog/datadog-agent/pull/18592

**Pull request closed without merging**